File: templates/native-android/app/src/main/java/com/example/MainActivity.java
package {{PACKAGE_NAME}};

import android.app.Activity;
import android.os.Bundle;
import android.webkit.WebView;

public class MainActivity extends Activity {
    @Override
    protected void onCreate(Bundle savedInstanceState){
        super.onCreate(savedInstanceState);
        WebView w = new WebView(this);
        w.getSettings().setJavaScriptEnabled(true);
        w.loadUrl("file:///android_asset/www/index.html");
        setContentView(w);
    }
}

File: .workspace/zip-build-1787509026079/app/src/main/java/com/callrecorder/ry/ui/main/MainActivity.kt
package com.callrecorder.ry.ui.main
 
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.NavController
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.ui.setupWithNavController
import com.callrecorder.ry.R
import com.callrecorder.ry.databinding.ActivityMainBinding
import com.callrecorder.ry.service.CallDetectionReceiver

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private lateinit var navController: NavController

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setupNavigation()
        CallDetectionReceiver.registerTelephonyListener(this)
    }

    private fun setupNavigation() {
        val navHostFragment = supportFragmentManager
            .findFragmentById(R.id.nav_host_fragment) as? NavHostFragment
        if (navHostFragment != null) {
            navController = navHostFragment.navController
            binding.bottomNavigation.setupWithNavController(navController)
        }
    }
}
